src/number_format.js

```diff
@@ -671,6 +671,15 @@ class NumberFormat extends React.Component {
       );
     }
 
+    /** Check for "paste event" with, replace any allowed decimal separator with desired decimal separator. Issue #349  */
+    if (!format && decimalScale !== 0 && allowedDecimalSeparators.some(separator => value.indexOf(separator) !== -1)) {
+      let result = value;
+      allowedDecimalSeparators.forEach(v => {
+        result = result.replace(v, decimalSeparator);
+      })
+      value = result;
+    }
```

Comment:
- The logic can be simplifed using regex, instead of array.
```
const decimalSeparatorRegex = new RegExp(allowedDecimalSeparators.join('|'));
```
- If we add this the previous `if` condition is no longer needed.
- The replace method should happen only on characters between prefix and suffix. What if prefix or suffix has a decimal separator character. like suffix `.sqft`. Can we add a spec to verify that? <- Add decimal char on the prefix, as replace replaces first char.

Reply:
Also, while looking into code, I feel this transformation is non needed to be done on `correctInputValue`. Instead we should do it in formatInput method, after this.
https://github.com/s-yadav/react-number-format/blob/master/src/number_format.js#L621

That way we don't have to worry about prefix and suffix. Also, we can remove transformation for allowedDecimalSeparator in line 663.
https://github.com/s-yadav/react-number-format/blob/master/src/number_format.js#L663

Reply:
correctInputValue, is mostly a supporting function for keyDown event. This was introduced due to android keyboard bug, where it doesn't give the correct character code on the keyDown event. Not sure if it still exist.

Reply:
On further discussion looks like the paste event is tricky, what if the pasted text have a thousand separator on it. For example, `1,111.11` This logic will break.

Also, how do you identify if `,` in pasted character is supposed to be thousand separator or decimal separator. For example what we should treat `,` in this. `1,111`.?

Reply:
> On further discussion looks like the paste event is tricky, what if the pasted text has a thousand separator on it. For example, `1,111.11` This logic will break.
> 
> Also, how do you identify if `,` in pasted character is supposed to be thousand separator or decimal separator. For example what we should treat `,` in this. `1,111`.?

As far as i know you cannot have allowedDecimalSeparator same as thousand separator. I believe decimal separator difference is more common than thousand separator differences.
When you look at the current library props it looks like you can enter different decimal place separator, but only one thousand separator - and that should stay that way

Reply:
Hey, just one small clarification on this _(assuming this is what you meant)_

> As far as i know you cannot have allowedDecimalSeparator same as thousand separator. I believe decimal separator difference is more common than thousand separator differences.

At the moment, we cannot have decimal separator same as the thousand separator. Here is a small demo:           https://codesandbox.io/s/react-number-format-demo-forked-gpm7s
It should throw this error: `Decimal separator can't be same as thousand separator.`

This, -on the other hand, is possible

```
 <NumberFormat
        allowedDecimalSeparators={[",", "."]}
        decimalSeparator="."
        isNumericString
        thousandSeparator=","
      />
```

For some more detail on `allowedDecimalSeparators` you can refer this: #345

Reply:
Do we really want to support this configuration with thousandSeparator=',' and allowedDecimalSeparators={[",", "."]} ?

Reply:
I think there is valid usecase for that configuration.
https://github.com/s-yadav/react-number-format/issues/324#issuecomment-534439760

Reply:
Other people also to support similar usecase. If you go through through linked issues in this PR. #345 

The allowedDecimalSeparator was introduced to aid on keyboards where locale decides what char to show. Paste is tricky as mentioned above.

One thing we can do is to run the logic only when thousandSeparator doesn't conflict with allowdDecimalSeparator.

This will not solve the problem completely but partial improvement is better than no improvement.

Thoughts @Toumash, @nikhil-varma

Reply:
I think considering what `allowDecimalSeparator` is doing, a validation agains `thousandSeparator` might be a good idea. On that note, there are one of two things we can choose to do
- Add a base check to ensure that `thousandSeparator` is not included in `allowDecimalSeparator` _(Rationale is basically that a string cannot have the same radix character and thousand separator `_ , Ideally _(in the context of our library)_ the radix character is dependent on either
  - The value of `decimalSeparator` 
  - Default `.`
- Just consider this check while pasting 

I am guessing adding it as a base condition is a much cleaner idea because the library will be able to tell the users' that it is not a valid config and will provide a clear segregation unless we have a use-case where it might fail. Because in the end, a value in the `thousandSeparator` is an allowed character but not as a radix character. 

What do you think @s-yadav @Toumash

Reply:
Validation might not be possible as there is a valid usecase for the conflict.

https://github.com/s-yadav/react-number-format/issues/324#issuecomment-534439760

Reply:
Oh yeah you are right! But then if we just add it in this particular context, this specific use-case might fail? 🤔

Reply:
Okay, maybe we should just add this workaround into the README.md then?

```
const DemoField =  useMemo(() => NumberFormatCommaPasteHackTextField(setValue), []);

const NumberFormatCommaPasteHackTextField = (setValue) =>
  (props) => {
    return <TextField
      {...props}
      onPaste={(e) => {
        let pastedText = e.clipboardData.getData('text');
        if (pastedText.indexOf(',') !== -1) {
          e.preventDefault();
          setValue(pastedText.replace(',', '.'));
        }
      }}
    />
  }

return  <NumberFormat
        customInput={DemoField}
        allowedDecimalSeparators={[",", "."]}
        decimalSeparator={'.'}
        thousandSeparator=" "
      />
```

Reply:
Adding it to the documentation may be an issue because it would become a suggested approach and the actual issue won't be resolved. 
Let's do this
- Change documentation for `allowedDecimalSeparators` saying that paste may not work as expected if there is a conflict between `thousandSeparator` and `allowedDecimalSeparators`. Let's also link this PR there in the documentation to provide the rationale behind it.
- We can add a conflict check in your logic between `thousandSeparator` and `allowedDecimalSeparators` as @s-yadav pointed out. 

If there is no conflict then the paste will work as expected but if there is it may not as suggested above.

Reply:
Okay, gonna provide a new version in the next days